### PR TITLE
DataGrid add property to specify Number Format of items count

### LINKED ISSUE #14464: DataGrid row group header items count format should be easier to change
## Is your feature request related to a problem? Please describe.
Items count number should by easly formatable to 123 456 not only default 123456 format

## Describe the solution you'd like
Add property to DataGridRowGroupHeader

## Additional context
At this point is still posible to do this but mayby this In my opinion this should be done easier..

https://github.com/AvaloniaUI/Avalonia/blob/18d02f1aa005b87ff85c995751b73e02f7e38567/src/Avalonia.Controls.DataGrid/DataGridRowGroupHeader.cs#L446-L450

image

vs 

image

### PR DESCRIPTION
**Pull request by KrzysztofDusko**
## What does the pull request do?
DataGrid add property to specify Number Format of items count.
**edit** - after sugestions scope of this issue is to cover not only number but whole header text

## What is the current behavior?
format cant be change easly

## What is the updated/expected behavior with this PR?
format can be changed on LoadingRowGroup event

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/avalonia-docs with user documentation

## Fixed issues
Fixes  #14464

### CODE REVIEW

File: src/Avalonia.Controls.DataGrid/DataGridRowGroupHeader.cs [hunk 1]
```diff
@@ -430,6 +430,10 @@ internal void ToggleExpandCollapse(bool isVisible, bool setCurrent)
             }
         }
 
+        /// <summary>
+        /// Gets or sets a value that indicates number format of items count
+        /// </summary>
+        public string CountFormat { get; set; } = "N1";
```
Comment (workgroupengineering) [inline src/Avalonia.Controls.DataGrid/DataGridRowGroupHeader.cs:436]: make it a StyledProperty so you can customize via styles or ControlTheme

File: src/Avalonia.Controls.DataGrid/DataGridRowGroupHeader.cs [hunk 2]
```diff
@@ -444,10 +448,10 @@ internal void UpdateTitleElements()
             if (_itemCountElement != null && RowGroupInfo != null && RowGroupInfo.CollectionViewGroup != null)
             {
                 string formatString;
-                if(RowGroupInfo.CollectionViewGroup.ItemCount == 1)
+                if (RowGroupInfo.CollectionViewGroup.ItemCount == 1)
                     formatString = "({0} Item)";
                 else
-                    formatString = "({0} Items)";
+                    formatString = $"({{0:{CountFormat}}} Items)";
```
[the remark this answers is not in the document]
Comment (KrzysztofDusko) [reply src/Avalonia.Controls.DataGrid/DataGridRowGroupHeader.cs]: thanks for sugestion, applied

File: src/Avalonia.Controls.DataGrid/DataGridRowGroupHeader.cs (hunk 1 above)
Comment (KrzysztofDusko) [reply src/Avalonia.Controls.DataGrid/DataGridRowGroupHeader.cs]: thanks for sugestion, applied

**Comment by KrzysztofDusko** [comment]
suggestions apllied and i change name to ItemCountFormat to be consistent with IsItemCountVisible

**Comment by maxkatz6** [comment]
Alternatively, we could have a IBinding property, similarly to this one: https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Primitives/SelectingItemsControl.cs#L250-L252

But I might overcomplicate things here.

**Review by maxkatz6 — APPROVED** [review]
Thank you!

**Comment by KrzysztofDusko** [comment]
> Thank you!

somethigs goes wrong ? I dont see changes in master :(

**Pull request merged**